webpack/assets/javascripts/react_app/components/FactCharts/FactChartActions.js

```diff
@@ -1,23 +1,16 @@
 import {
-  FACT_CHART_REQUEST,
-  FACT_CHART_SUCCESS,
-  FACT_CHART_FAILURE,
+  FACT_CHART,
   FACT_CHART_MODAL_OPEN,
   FACT_CHART_MODAL_CLOSE,
 } from './FactChartConstants';
+import { API_OPERATIONS } from '../../redux/API';
 
-import { ajaxRequestAction } from '../../redux/actions/common/index';
-
-export const getChartData = (url, id) => dispatch =>
-  ajaxRequestAction({
-    dispatch,
-    requestAction: FACT_CHART_REQUEST,
-    successAction: FACT_CHART_SUCCESS,
-    failedAction: FACT_CHART_FAILURE,
-    url,
-    item: { id },
-  });
-
+export const getChartData = (url, id) => ({
+  type: API_OPERATIONS.GET,
+  outputType: FACT_CHART,
```

Comment:
Can you rename `outputType` with `key` ?
In another PR I wanted to add functionality to the API action,
I wanted to access the `outputType` and there it feels out of context..
`key` also indicates that this API process is unique, to prevent people from overriding it.

Reply:
Renamed and rebased